Makefile

```diff
@@ -22,12 +22,19 @@ test: test-versions test-installed test-coverage
 	@echo "\n(test) completed\n"
 
 test-local:
-	@pip3 install -r requirements-optional.txt
+	@pip3 install --upgrade -e .[dev]
 	@make test-installed
 	make test-versions
 	make test-coverage
 	@# test-coverage needs to be last in deps, don't clean after it runs!
 
+test-on-containers: clean-silent
+	@# Run the tests on pristine containers to isolate us from the local environment.
+	@for version in 3.8 3.9 3.10 3.11 3.12.0; do \
```

Comment:
I specify 3.12.0 for now as we are still incompatible with 3.12.1. I want to make testing easier before pushing the compatibility fixes for 3.12.1.